src/EnergyPlus/WeatherManager.cc

```diff
@@ -6111,8 +6111,15 @@ namespace WeatherManager {
             if (RunPeriodInput(Loop).startMonth == 2 && RunPeriodInput(Loop).startDay == 29) {
                 // Requested start date is a leap year
                 if (RunPeriodInput(Loop).startYear == 0) { // No input starting year
-                    RunPeriodInput(Loop).startYear =
-                        findLeapYearForWeekday(RunPeriodInput(Loop).startMonth, RunPeriodInput(Loop).startDay, RunPeriodInput(Loop).startWeekDay);
+                    if (inputWeekday) {
+                        RunPeriodInput(Loop).startYear =
+                            findLeapYearForWeekday(RunPeriodInput(Loop).startMonth, RunPeriodInput(Loop).startDay, RunPeriodInput(Loop).startWeekDay);
+                    } else {
+                        // 2012 is the default year, 1/1 is a Sunday
+                        RunPeriodInput(Loop).startYear = 2012;
+                        RunPeriodInput(Loop).startWeekDay =
+                            calculateDayOfWeek(RunPeriodInput(Loop).startYear, RunPeriodInput(Loop).startMonth, RunPeriodInput(Loop).startDay);
+                    }
```

Comment:
@mjwitte Test 1 sounds like expected behavior since we're not using the end date to set the start date (so it'll run from 2017 to 2020). Test 3 should be similar, it'll try to run from 2007 through 2/29/2008. Were you using our Tampa EPW for 2 and 4? If so, that's what should happen since there's no 2/29 in the file. Will run some tests locally.

Reply:
Oh I see. Didn't think about what happens if the epw doesn't have Feb 29. It would be nice to say something to that effect in the error message.

Reply:
Actually, in this case, it should run through 2/28 in both test 1 and 3 since there's no 2/29. I've modified the error message to be:

```
   ** Severe  ** Multiple rewinds on EPW while searching for first day 2/29
   **  Fatal  ** Error occurred on EPW while searching for first day, stopped at 2004/12/31 1:0 IO Error=-1
```

or:

```
   ** Severe  ** Multiple rewinds on EPW while searching for first day 2/29/1992
   **  Fatal  ** Error occurred on EPW while searching for first day, stopped at 2004/12/31 1:0 IO Error=-1
```
depending upon whether the weather is actual or not. I also put in a space in between the date and the time so it's a little easier to read.